src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -342,11 +432,20 @@ private void validateSchema(String tableName, Schema tableSchema) throws IOExcep
     return Objects.requireNonNull(tableSchema.getFields()).stream()
       .map(field -> new BigQueryTableFieldSchema()
         .setName(field.getName())
-        .setType(getTableDataType(BigQueryUtil.getNonNullableSchema(field.getSchema())).name())
-        .setMode(Field.Mode.NULLABLE.name()))
+        .setType(getTableDataType(field.getSchema()).name())
+        .setMode(getMode(field.getSchema()).name()))
       .collect(Collectors.toList());
   }
 
+  private Field.Mode getMode(Schema schema) {
+    if (BigQueryUtil.getNonNullableSchema(schema).getComponentSchema() != null) {
```

Comment:
it's better if this is a check that the schema type is array. Though these two are logically equivalent today, it is more clear if this is a type check on array.